inspire_utils/urls.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""URL-related utils."""
+
+from __future__ import absolute_import, division, print_function
+
+import re
+
+from six import text_type
+
+
+def ensure_scheme(url, default_schema='http'):
```

Comment:
don't do this by hand, look at `urlsplit` and `urljoin` from [`urlparse`](https://docs.python.org/2/library/urlparse.html) imported as `six.moves.urllib.parse`: you should be able to do something like `urljoin(urlsplit(url, scheme='http'))`.

Reply:
There is an interesting thing happening when I use this approach:

```python
In [11]: urlunsplit(urlsplit('http://inspirehep.net', scheme='https'))
Out[11]: 'http://inspirehep.net'

In [12]: urlunsplit(urlsplit('inspirehep.net', scheme='https'))
Out[12]: 'https:///inspirehep.net'
```

In the cases where schema is not provided, `inspirehep.net` is treated as a `path`, instead of `netloc`, because urlparse expects the netloc to start with '//'. I found [this potential solution](https://stackoverflow.com/a/2499088), however I'm not sure if in the end it's cleaner?